### ISSUE TITLE
haskell_module emits bogus warnings when using plugins

### ISSUE THREAD

**Issue by facundominguez** [tweag/rules_haskell#1681]
When a `haskell_module` rule uses plugins, and the rule doesn't enable TH, warnings of the following form are printed in the console:
```
INFO: From HaskellBuildObject //tests/haskell_module/plugin:lib //tests/haskell_module/plugin:module-with-plugin:

<no location info>: warning: [-Wmissed-extra-shared-lib]
    libHSexceptions-0.10.4.so: cannot open shared object file: No such file or directory
    It's OK if you don't want to use symbols from it directly.
    (the package DLL is loaded by the system linker
     which manages dependencies by itself).

<no location info>: warning: [-Wmissed-extra-shared-lib]
    libHSsplitmix-0.1.0.4.so: cannot open shared object file: No such file or directory
    It's OK if you don't want to use symbols from it directly.
    (the package DLL is loaded by the system linker
     which manages dependencies by itself).

<no location info>: warning: [-Wmissed-extra-shared-lib]
    libHSrandom-1.2.0.so: cannot open shared object file: No such file or directory
    It's OK if you don't want to use symbols from it directly.
    (the package DLL is loaded by the system linker
     which manages dependencies by itself).

<no location info>: warning: [-Wmissed-extra-shared-lib]
    libHStemporary-1.3.so: cannot open shared object file: No such file or directory
    It's OK if you don't want to use symbols from it directly.
    (the package DLL is loaded by the system linker
     which manages dependencies by itself).
```

The cause of these messages is that loading plugins requires the internal interpreter, which in turn wants to load the shared libraries of the packages in the dependencies. However, when TH is not enabled, those shared libraries are not provided as inputs to the build action, thus producing the above warnings.

This state of affairs should not cause build failures, because when the plugin needs some shared library, it will be listed in the dependencies of the plugin, and these plugin dependecies are fed as inputs even with TH disabled.

This can be reproduced with [these changes](https://github.com/tweag/rules_haskell/compare/fd/plugins-warnings) and then running
```
nix-shell --pure --run "bazel build //tests/haskell_module/plugin:lib"
```
### Expected behavior

I can think of the following fixes
1. Have `rules_haskell` add `-Wno-missed-extra-shared-lib` to the compiler flags when using plugins and TH is disabled.
2. Ask the user to suppress the warnings with `-Wno-missed-extra-shared-lib` when using plugins

Perhaps (1) is the way to go. I don't know in which cases the user would want to have the warning enabled other than debugging `rules_haskell` itself.
